src/software/new_geom/convex_polygon.cpp

```diff
@@ -39,7 +39,9 @@ bool ConvexPolygon::isConvex()
         totalAngle += (180 - vertexAngleInDegrees);
     }
 
-    return std::abs(totalAngle - 360) < GeomConstants::EPSILON;
+    // TODO: Some polygons that SHOULD be convex aren't passing this with
```

Comment:
```GeomConstants::EPSILON``` is not tolerant enough for some convex polygons. For example, ```Triangle((Point(0, 0), Point(4.5, 0.5), Point(4.5, -0.5))``` should be a valid convex polygon but it is failing ```isConvex(...)``` check with our current EPSILON, I changed it to ```1e-12``` for now but I'm not sure how we should go about this? Is a different ```EPSILON``` for this okay?

Reply:
So looking at the implementation of `isConvex`, I'm not super surprised we hit this issue. On each corner angle calculation we're going to accumulate some floating point error, which could easily make the final check fail. I think we probably want to change over to a numerically stable implementation. 

I'm fairly sure just checking that:
1) abs(each corner) is < 180 deg
2) All angles are either greater or less then zero (depending if points are given in CW or CCW order, _do we have a test CW and CCW?_)
3) The polygon is not self-intersecting (this effectively replaces the final check)

Reply:
I'll try changing the implementation of ```isConvex``` and see how that goes

Reply:
"The polygon is not self-intersecting"

Looked into this and it isn't trivial to implement. What I did for now is to change any use of degrees in ```isConvex``` to radians and it seems to be more precise as I was able to lower the EPSILON used. Maybe in a different issue, we can have a proper implementation of ```isConvex``` that uses what you mentioned above?

Reply:
I'm totally fine if we open an issue here and put it off (as long as we prioritize it), but out of curiosity, what is the complexity with self intersections? Wouldn't it basically be:
```
for (Edge e1 : edges){
     for (Edge e2 : edges){
        if (intersects(e1, e2))  {
             return true
        }
    }
}
return false;
```

Reply:
Regarding the above pseudocode, just need to be careful about your edges touching at the vertices and whether `intersects` returns `true` in that case or not. Possible "interface complexity" instead of time complexity.

Reply:
I was thinking of implementing the bentley ottman algorithm to avoid having ```O(n^2)``` but that algorithm isn't trivial. If we're happy with your pseudocode then it should be fine given Mathew's comment - I can check in my next update

Reply:
Yeah, I agree `O(n^2)` is not ideal, but I think for the vast majority of our use cases right now it'd be totally fine. Bentley-Ottoman would definitely be much better though, maybe just a file an issue so we don't forget to come back and change it!

Reply:
I tried to do a simple implementation of above pseudocode but since geom/util still has intersects and geom/util now depends on new geom shape hierarchy, we end up with a circular dependency and the build fails. I think I'm going to keep it as is for now and file an issue to fix it later if that's okay with you guys @garethellis0 @MathewMacDougall

Reply:
Yeah ofc. Just file a ticket with everything you have so far, stuff discussed here, anything you've implemented so we don't lose info/work. 

Sorry about this, really didn't mean to hold up this PR. :grimacing:

Reply:
No problem, filed ticket #1140 for this